build-caching/src/main/java/com/vertispan/j2cl/build/task/Dependency.java

```diff
@@ -1,10 +1,19 @@
 package com.vertispan.j2cl.build.task;
 
+import java.io.File;
+import java.util.Set;
+
 public interface Dependency {
     Project getProject();
 
     Scope getScope();
 
+    boolean isAPT();
+
+    File getJar();
+
+    Set<String> getProcessors();
```

Comment:
Can we move these new methods to Project instead of Dependency? Also please qualify them with comments like isJsZip() has, that this might be temporary, other warnings on usage.